app/controllers/api/v1/documents_controller.rb

```diff
@@ -24,25 +24,27 @@ def destroy
 
     private
 
-      PERMITTED_DOCUMENT_PARAMS = [
-        attributes: %i[file],
-      ].freeze
-
-      def document_params
-        params.require(:data).permit(PERMITTED_DOCUMENT_PARAMS).to_h
+      def document_attributes
+        document_params[:attributes].merge(documentable: profile)
       end
 
-      def document_attributes
-        document_params[:attributes].merge(
-          move: Move.find_by(id: params.dig(:move_id)),
-        )
+      def document_params
+        params.require(:data).permit(attributes: %i[file]).to_h
       end
 
     protected
 
       def set_restricted_request_content_type
         @restricted_request_content_type = 'multipart/form-data'
       end
+
+      def profile
+        move_id = params[:move_id]
+
+        if move_id
+          Move.accessible_by(current_ability).find(move_id)&.profile
```

Comment:
Do we need `&.` to get the profile?

that is part of the Move if it's empty you'd get nil,
it the move is not found it raises an exception

Reply:
We did need it, yeah